### ISSUE TITLE
Build failed in mina.netty: NioDatagramConnectorExIT

### ISSUE THREAD

**Issue by cmebarrow** [kaazing/gateway#67]
[Build 60](https://travis-ci.org/kaazing/gateway/builds) failed on oracle and open jdk with the following test failing in mina.netty:
````
Running org.apache.mina.transport.socket.nio.NioDatagramConnectorExIT
Tests run: 1, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 0.468 sec <<< FAILURE! - in org.apache.mina.transport.socket.nio.NioDatagramConnectorExIT
shouldConnect(org.apache.mina.transport.socket.nio.NioDatagramConnectorExIT)  Time elapsed: 0.426 sec  <<< FAILURE!
java.lang.AssertionError: not all expectations were satisfied
expectations:
  expected once, already invoked 1 time: ioHandler.sessionCreated(instance of org.kaazing.mina.core.session.IoSessionEx)
  expected once, already invoked 1 time: ioHandler.sessionOpened(instance of org.kaazing.mina.core.session.IoSessionEx)
  ! expected once, never invoked: ioHandler.sessionClosed(instance of org.kaazing.mina.core.session.IoSessionEx)
what happened before this:
  ioHandler.sessionCreated(<(0x00000001: nio datagram, client, /127.0.0.1:52995 => /127.0.0.1:2101)>)
  ioHandler.sessionOpened(<(0x00000001: nio datagram, client, /127.0.0.1:52995 => /127.0.0.1:2101)>)
	at org.jmock.lib.AssertionErrorTranslator.translate(AssertionErrorTranslator.java:20)
	at org.jmock.Mockery.assertIsSatisfied(Mockery.java:199)
	at org.apache.mina.transport.socket.nio.NioDatagramConnectorExIT.after(NioDatagramConnectorExIT.java:63)
````

**Comment by cmebarrow** [comment]
I was successfully able to build on my machine just now, so apparently this is a sporadic failure, probably not caused by the previous commit.

**Comment by cmebarrow** [comment]
This is probably happening because the org.kaazing.mina.core.filterchain.DefaultIoFilterChain.fireSessionClosed() method:
{code:java}
    public void fireSessionClosed() {
        // Update future.
        try {
            session.getCloseFuture().setClosed();
        } catch (Throwable t) {
            fireExceptionCaught(t);
        }

        // And start the chain.
        Entry head = this.head;
        callNextSessionClosed(head, session);
    }
{code}
sets the close future to closed before firing this session closed event on the filter chain which will cause handler.sessionClosed to be called (from the Tail filter), so the expectation for handler.sessionClosed may not be fulfilled if the test completes before that method gets invoked.
:

**Comment by cmebarrow** [comment]
Fixed by pull request #74 which has been merged.
